File: client/src/main/java/com/jraska/github/client/push/PushTokenService.java
package com.jraska.github.client.push;

import com.google.firebase.iid.FirebaseInstanceIdService;
import com.jraska.github.client.GitHubClientApp;

public final class PushTokenService extends FirebaseInstanceIdService {
  @Override public void onTokenRefresh() {
    GitHubClientApp app = ((GitHubClientApp) getApplication());
    app.pushHandler().onTokenRefresh();
  }
}

File: client/src/main/java/com/jraska/github/client/push/PushTokenService.kt
package com.jraska.github.client.push

import com.google.firebase.iid.FirebaseInstanceIdService
import com.jraska.github.client.GitHubClientApp

class PushTokenService : FirebaseInstanceIdService() {
  override fun onTokenRefresh() {
    val app = application as GitHubClientApp
    app.pushHandler().onTokenRefresh()
  }
}
